server/sttp-mock-server/src/main/scala/sttp/tapir/server/mockserver/impl/JsonCodecs.scala

```diff
@@ -81,12 +83,17 @@ private[mockserver] object JsonCodecs {
       }
   }
 
-  private implicit val expectationRequestDefinitionCodec: Codec[ExpectationRequestDefinition] = deriveCodec[ExpectationRequestDefinition]
-  private implicit val expectationResponseDefinitionCodec: Codec[ExpectationResponseDefinition] = deriveCodec[ExpectationResponseDefinition]
-  private implicit val expectationTimesCodec: Codec[ExpectationTimes] = deriveCodec[ExpectationTimes]
-  private implicit val expectationTimeToLiveCodec: Codec[ExpectationTimeToLive] = deriveCodec[ExpectationTimeToLive]
+  private def deriveCodecDropNull[A](implicit codec: Lazy[DerivedAsObjectCodec[A]]): Codec[A] =
+    Codec.from(codec.value, codec.value.mapJson(_.dropNullValues))
```

Comment:
doesn't this fix re-introduce https://github.com/softwaremill/tapir/issues/2128 ?

Reply:
🤦 ok I've read the code, that's a completely different thing. Looks good :)

Reply:
@adamw nope, all the tests pass; 

this one is intentionally non-recursive drop nulls, by the mockserver contract we need to drop nulls in the _request_ body in case there are nulls, but we don't want to drop nulls in the user body if it's not null;

So any httpRequest / httpResponse non recursive nulls should be dropped:

Invalid JSON: 

```js
{
  "httpRequest": {
    "method": "GET",
    "path": "/api/v1/order",
    "body": null, // we need to drop null if the body is null
    "headers": {
      "Accept-Encoding": [
        "gzip, deflate"
      ]
    }
  },
  "httpResponse": ...
}
```

Valid JSONs:

```js
{
  "httpRequest": {
    "method": "GET",
    "path": "/api/v1/order",
    "headers": {
      "Accept-Encoding": [
        "gzip, deflate"
      ]
    }
  },
  "httpResponse": ...
}
```

or 

```js
{
  "httpRequest": {
    "method": "GET",
    "path": "/api/v1/order",
    "body": { "field": null },
    "headers": {
      "Accept-Encoding": [
        "gzip, deflate"
      ]
    }
  },
  "httpResponse": ...
}
```